base/sparse/linalg.jl

```diff
@@ -532,11 +531,14 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = typeof(float(zero(T)))
+
+    S = zeros(Ti, n, t)
 
     # Generate the block matrix
-    X = Array(T, n, t)
-    X[1:n,1] = 1
+    X = Array(Ti, n, t)
+    X[1:n,1] = one(Ti)
```

Comment:
This is not really necessary because `1` will costlessly be converted to the element type of `X`.